server/parser/src/main/java/org/eclipse/lsp/cobol/AntlrRangeUtils.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2024 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *    Broadcom, Inc. - initial API and implementation
+ *
+ */
+package org.eclipse.lsp.cobol;
+
+import org.antlr.v4.runtime.ParserRuleContext;
+import org.antlr.v4.runtime.Token;
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.Range;
+/**
+ * The AntlrRangeUtils class provides a utility method to construct Range objects.
+ */
+public class AntlrRangeUtils {
+    /**
+     * Construct the range from ANTLR context
+     *
+     * @param ctx the ANTLR context
+     * @return the range
+     */
+    public static Range constructRange(ParserRuleContext ctx) {
+      Token start = ctx.start;
+      Token stop = ctx.stop == null ? start : ctx.stop;
```

Comment:
```java
if (ctx.stop == null || "start > stop") {
  Position pos = new Position(start.getLine() - 1, start.getCharPositionInLine());
  return new Range(pos, pos);
}
```
otherwise you'll get a range with one char length.